spec/lib/extensions/brakeman_fingerprint_patch_spec.rb

```diff
@@ -0,0 +1,168 @@
+        let(:engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "vendor/bundle/ruby/3.1.5/bundler/gems/manageiq-ui-classic-0.1.0/app/controllers/application_controller.rb",
+            :absolute => "/Users/user/dev/manageiq/vendor/bundle/ruby/3.1.5/bundler/gems/manageiq-ui-classic-0.1.0/app/controllers/application_controller.rb"
+          )
+        end
+
+        include_examples "handles brakeman pathing"
+      end
+    end
+
+    context "with a path-based engine" do
+      context "which is a sibling of the core repo" do
+        let(:engine_path) { "/Users/user/dev/manageiq-ui-classic/" }
+        let(:engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "../manageiq-ui-classic/app/controllers/application_controller.rb",
+            :absolute => "/Users/user/dev/manageiq-ui-classic/app/controllers/application_controller.rb"
+          )
+        end
+
+        include_examples "handles brakeman pathing"
+      end
+
+      context "which is inside of the core repo" do
+        let(:engine_path) { "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/" }
+        let(:engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "plugins/manageiq-ui-classic/app/controllers/application_controller.rb",
+            :absolute => "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/app/controllers/application_controller.rb"
+          )
+        end
+
+        include_examples "handles brakeman pathing"
+      end
+    end
+  end
+
+  context "running from an engine repo" do
+    context "with a symlinked spec/manageiq dir" do # When symlinked, the paths appear the same as a path-based gem, so these are copies of the path-based tests above
+      context "which is a sibling of the core repo" do
+        let(:engine_path) { "/Users/user/dev/manageiq-ui-classic/" }
+        let(:engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "../manageiq-ui-classic/app/controllers/application_controller.rb",
+            :absolute => "/Users/user/dev/manageiq-ui-classic/app/controllers/application_controller.rb"
+          )
+        end
+
+        include_examples "handles brakeman pathing"
+      end
+
+      context "which is inside of the core repo" do
+        let(:engine_path) { "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/" }
+        let(:engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "plugins/manageiq-ui-classic/app/controllers/application_controller.rb",
+            :absolute => "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/app/controllers/application_controller.rb"
+          )
+        end
+
+        include_examples "handles brakeman pathing"
+      end
+    end
+
+    context "with a cloned spec/manageiq dir" do # This is also the CI case for a plugin
```

Comment:
@jrafanie When I ran these tests without the code fix, all of them passed except this one, and within this one the engine one passed but the core case did not.  So, the code fix specifically fixes this case without breaking the other cases.